daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v2/ScriptF.scala

```diff
@@ -778,6 +778,42 @@ object ScriptF {
       )
   }
 
+  final case class VetDar(
+      darName: String,
+      participant: Option[Participant],
+  ) extends Cmd {
+    override def execute(env: Env)(implicit
+        ec: ExecutionContext,
+        mat: Materializer,
+        esf: ExecutionSequencerFactory,
+    ): Future[SExpr] =
+      for {
+        client <- env.clients.getParticipant(participant) match {
+          case Right(client) => Future.successful(client)
+          case Left(err) => Future.failed(new RuntimeException(err))
+        }
+        _ <- client.vetDar(darName)
```

Comment:
Would it make sense to instead Await here, given that the vetDar on the admin API should be synchronous?

Reply:
The whole computation is a future, it should wait for the vetDar future to end before returning SUnit, no?